<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Validation error</title>
</head>
<body>
    <script>

        //class validation err
        class ValidationError extends Error{
            constructor(message, field){
                super(message)
                this.field = field
            }
        }

        class MathUtil {
            static sum(...numbers){

                if(numbers.length === 0){
                    //bukan new Error lagi tapi validationError kita bisa nambahin field "number"
                    throw new ValidationError("Total parameter harus lebih dari 0", "number")
                }

                let total = 0
                for(const number of numbers){
                    total += number
                }
                return total
            }
        }

        //update try catch instanceof validationError bisa ngambil field dan message nya apa
        try{
            console.info(MathUtil.sum())
            console.info("Eko")
        }catch(error){
            if(error instanceof ValidationError){
                console.info(`Terjadi error di field : ${error.field}, dengan error : ${error.message}`)
            }else{
                console.info("Program selesai")
            }
        }finally{
                console.info("Program selesai")
            }

            
        console.info("kannedy")

        
    </script>
</body>
</html>